src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,974 @@
+    {
+        protected abstract SyntaxNode GenerateExpressionFromOptionalParameter(IParameterSymbol parameterSymbol);
+        protected abstract SyntaxNode? GetLocalDeclarationFromDeclarator(SyntaxNode variableDecl);
+        protected abstract SyntaxNode UpdateArgumentListSyntax(SyntaxNode argumentList, SeparatedSyntaxList<SyntaxNode> arguments);
+
+        public sealed override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var (document, textSpan, cancellationToken) = context;
+            if (document.Project.Solution.Workspace.Kind == WorkspaceKind.MiscellaneousFiles)
+            {
+                return;
+            }
+
+            var expression = await document.TryGetRelevantNodeAsync<TExpressionSyntax>(textSpan, cancellationToken).ConfigureAwait(false);
+            if (expression == null || CodeRefactoringHelpers.IsNodeUnderselected(expression, textSpan))
+            {
+                return;
+            }
+
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var expressionType = semanticModel.GetTypeInfo(expression, cancellationToken).Type;
+            if (expressionType is null or IErrorTypeSymbol)
+            {
+                return;
+            }
+
+            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+
+            // Need to special case for expressions that are contained within a parameter
+            // because it is technically "contained" within a method, but an expression in a parameter does not make
+            // sense to introduce.
+            var parameterNode = expression.FirstAncestorOrSelf<SyntaxNode>(node => syntaxFacts.IsParameter(node));
+            if (parameterNode is not null)
+            {
+                return;
+            }
+
+            var generator = SyntaxGenerator.GetGenerator(document);
+            var containingMethod = expression.FirstAncestorOrSelf<SyntaxNode>(node => generator.GetParameterListNode(node) is not null);
+
+            if (containingMethod is null)
+            {
+                return;
+            }
+
+            var containingSymbol = semanticModel.GetDeclaredSymbol(containingMethod, cancellationToken);
+            if (containingSymbol is not IMethodSymbol methodSymbol)
+            {
+                return;
+            }
+
+            // Code actions for trampoline and overloads will not be offered if the method is a constructor.
+            // Code actions for overloads will not be offered if the method if the method is a local function.
+            var methodKind = methodSymbol.MethodKind;
+            if (methodKind is not (MethodKind.Ordinary or MethodKind.LocalFunction or MethodKind.Constructor))
+            {
+                return;
+            }
+
+            if (methodSymbol.Language.Equals("Visual Basic") && methodSymbol.Name.Equals("Finalize") && methodSymbol.IsOverride)
```

Comment:
lol. no.  

1. why are you filtering this out? 
2. dont' do a language speicif check in the base class. this is what abstract methods are for :)

Reply:
WellKnownMemberNames.DestructorName

Reply:
lol
1) because I want to keep it consistent with c# and destructors don't have parameters
2) okay
Is there a better way to check whether or not the method symbol is a destructor in the VB case without just comparing the method name?